lib/ProductOpener/Display.pm

```diff
@@ -6958,9 +6958,47 @@ sub display_field($$) {
 			}
 		}
 		my $lang_field = lang($field);
-		if ($lang_field eq '') {
+		if ($lang_field eq '' and $field ne 'states') {
 			$lang_field = ucfirst(lang($field . "_p"));
 		}
+		
+		# Separate To-Do and Done Status
+		if ($field eq 'states') {
+			my $done_status = '';
+			my $to_do_status = '';
+			my @status_split = split(',', $value);
+			foreach my $val (@status_split) {
+				if (index($val, "to-be") != -1) {
+		 			$to_do_status .=$val . ",";
+  		 		}
+		 		else {
+		 		$done_status .=$val . ",";
+		 		}
+			}
+			$to_do_status =~ s/,$//;
+			$done_status =~ s/,$//;
+			my $no_entry_to_do_status = $to_do_status;
+			my $no_entry_done_status = $done_status;
+			my $empty = '<a class="tag well_known" href="/state/empty">'."Empty".'</a>';
+			my $incomplete = '<a class="tag well_known" href="/state/to-be-checked">'."To be checked".'</a>';
+			# Regex to extract anchor tag content
+			$empty =~ s/<a[^>]*>|<\/a>//;
+			$incomplete =~ s/<a[^>]*>|<\/a>//;
+			$no_entry_done_status =~ s/<a[^>]*>|<\/a>//;
+			$no_entry_to_do_status =~ s/<a[^>]*>|<\/a>//;
+			if ($no_entry_to_do_status eq $incomplete){
+				$lang_field = "Done";
+				$value = $done_status;
+			}
+			else {
+				$lang_field = "To Do";
+				$value = $to_do_status;
+			}
+			if ($no_entry_done_status ne $empty and $no_entry_to_do_status ne $incomplete){
+				$html .= '<p><span class="field">' . "Done" . separator_before_colon($lc)  . ":</span>" . $done_status . "</p>";
+			}
+		}
+		
```

Comment:
Here you can add an else { 
$html .= '<p><span class="field">' . $lang_field . separator_before_colon($lc) . ":</span> $value</p>";
}

so that you don't print it again (you will already have printed the "Done" and "To do" values above)